### PR TITLE
Remove last direct use of UnmarshalKey in favor of pkg/config/structure

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Remove last direct use of UnmarshalKey in favor of pkg/config/structure

### Describe how to test/QA your changes

Functionality is currently guarded behind a env var, so there should be no functional change. To test the behavior, use the DD_CONF_NODETREEMODEL env var, but be warned that there are likely bugs.

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

File: pkg/config/structure/unmarshal_test.go [hunk 1]
```diff
@@ -40,6 +40,17 @@ type trapsConfig struct {
 	Namespace        string   `yaml:"namespace"`
 }
 
+// We don't use config mock here to not create cycle dependencies (same reason why config mock are not used in
+// pkg/config/{setup/model})
```
Comment (dustmop) [inline pkg/config/structure/unmarshal_test.go:44]: Cool, this explains well why this fix works to handle the issue I originally ran into when I last tried to use UnmarshalKey in the remaining sites.

**Review by dustmop — APPROVED** [review]
LGTM

**Review by jedupau — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]

**Review by moth-a — APPROVED** [review]
I don't believe RC files are touched, but lgtm!

**Review by hmahmood — APPROVED** [review]

**Review by mackjmr — APPROVED** [review]

**Pull request merged**